### DISCUSSION TITLE
Blink an output for x times via MQTT

### DISCUSSION THREAD

**Discussion by DarthWeber** [arendst/Tasmota#22086] [General]
Hi, maybe someone can help me with this feature i wish to realize:

I would like to send via http or mqtt a number and then tasmota should switch an output for this number times with 1 or 2s on and off time. So e.g. if i send "5" the output should do:

ON-OFF-ON-OFF-ON-OFF-ON-OFF-ON-OFF

with fixed ON/OFF times. I have no problem to send this with 5 individual timed MQTT requests and a pulsetimed output. But i am interested in a solution with rules or berry.

Thanks,
Tom

**Comment by sfromis** [comment]
If you configure `BlinkTime` and `BlinkCount`, the command `Power blink` will trigger such blinking.

Since you ask about Berry, here's a rudimentary example of a more flexible command, you can embellish on:
```
class blinker
  def init()
    tasmota.remove_cmd('Blinker')
    tasmota.add_cmd('Blinker', /cmd,idx,payload,data->self.blinkstart(cmd,idx,payload,data))
  end
  def blinkstart(cmd,idx,payload,data)
    import string
    var args=string.split(payload,',')
    tasmota.set_timer(0, /->self.blink(int(args[0])*2, int(real(args[1])*1000)))
    tasmota.resp_cmnd_done()
  end
  def blink(count, interval)
    tasmota.cmd('Power1 toggle')
    if count > 1
      tasmota.set_timer(interval, /->self.blink(count-1, interval))
    end
  end
end
blinker()
```
